packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -105,9 +104,7 @@ class TimelineData {
 
     if (!_shouldProcessTraceEvent(event)) return;
 
-    if (debugTimeline) {
-      debugTraceEvents.write('${jsonEncode(event.json)},');
-    }
+    debugTraceEvents.write('${jsonEncode(event.json)},');
```

Comment:
similarly, just build up a Map here instead of a StringBuffer as it will be easier to manipulate and cheaper for the typical case that the data is accumulated but not actually used.